src/Pages/Withdrawals.razor

```diff
@@ -843,6 +907,115 @@
+
+            if (_selectedUTXOs.Count > 0)
+            {
+                await CoinSelectionService.LockUTXOs(_selectedUTXOs, newRequest, BitcoinRequestType.WalletWithdrawal);
+            }
+
+            _utxoSelectorModalRef.ClearModal();
+        }
+        else
+        {
+            ToastService.ShowError("Something went wrong");
+            _userPendingRequests.Remove(newRequest);
+        }
+    }
+
+    private async Task Bumpfee(WalletWithdrawalRequest request)
+    {
+        try {
+            await CreateNewWithdrawal(request);
+        } catch {
+            ToastService.ShowError("Could not bump the withdrawal");
+        }
+    }
+
+    private async void SubmitBumpfeeModal(WalletWithdrawalRequest newRequest)
+    {
+        if (newRequest != null && _bumpfeeRef != null) {
+            newRequest.WalletId = _selectedRequest.WalletId;
+            newRequest.Wallet = await WalletRepository.GetById(newRequest.WalletId);
+                
+            // Use either the custom fee rate or the recommended fee rate based on mempool fee type
+            decimal newFeeRate;
+            if (newRequest.MempoolRecommendedFeesType == MempoolRecommendedFeesType.CustomFee && newRequest.CustomFeeRate.HasValue)
+            {
+                newFeeRate = (decimal)newRequest.CustomFeeRate.Value;
+            }
+            else
+            {
+                // Use the recommended fee rate from NBXplorer service based on the selected mempool fee type
+                var recommendedFeeRate = await GetRecommendedFeeRate(newRequest.MempoolRecommendedFeesType);
+                newFeeRate = (decimal)recommendedFeeRate;
+            }
+
+            // If the new fee plus the amount exceeds the utxo sum value, show an error
+            // NOTE: revisit this when full rbf because we could add a UTXO
+            var txInfo = await NBXplorerService.GetTransactionAsync(uint256.Parse(_selectedRequest.TxId));
+            var tx = txInfo.Transaction;
+            long vSize = tx.GetVirtualSize();
+            decimal newFee = (vSize * newFeeRate) / 100_000_000m;
+
+            // Retrieve from database to get the UTXOs
+            _selectedRequest = await WalletWithdrawalRequestRepository.GetById(_selectedRequest.Id);
+
+            if (_selectedRequest == null)
+            {
+                ToastService.ShowError("Withdrawal request not found");
+                return;
+            }
+
+            if (_selectedRequest.UTXOs != null && _selectedRequest.UTXOs.Count > 0)
+            {
+                var inputAmount = Money.Satoshis(_selectedRequest.UTXOs.Sum(x => x.SatsAmount)).ToDecimal(MoneyUnit.BTC); // Convert to BTC
+                if (inputAmount <= newFee + _selectedRequest.TotalAmount)
+                {
+                    ToastService.ShowError("The new fee plus the amount exceeds the sum of the selected UTXOs. Please lower the fee rate.");
+                    return;
+                }
+            }
```

Comment:
should we put a lower max limit?

Reply:
it would be a good idea to add a estimate fees since we have the vbs of the tx, and also the % of the tx that goes to fees, will do it

Reply:
although I have set the limit to avoid dust, the rest is free to the user to do what they want

Reply:
done!